tests/testutil/service.go

```diff
@@ -202,6 +203,98 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+// GetEndpointURL returns the URL for this service's app if the App object is versioned it returns
+// the versioned url.
+func (ts *TestService) GetEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+// GetKubeEndpointUrlByRevision returns the versioned endpoint URL by using the revision provided
+// and returns it as string
+func (ts *TestService) GetKubeEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	args := []string{"--format", "json",
+		fmt.Sprintf("%s:%s",
+			strings.Split(ts.Name, ":")[0],
+			strings.Split(ts.Name, ":")[1])}
+	out, err := RioCmd("inspect", args)
+	s := riov1.Service{}
+	if err := json.Unmarshal([]byte(out), &s); err != nil {
+		return ""
+	}
+	args = []string{"get", "service.rio.cattle.io",
+		"-n", testingNamespace,
+		s.Name,
+		"-o", `jsonpath="{.status.endpoints[0]}"`}
+	url, err := KubectlCmd(args)
+	if err != nil {
+		ts.T.Fatalf("Failed to get endpoint url:  %v", err.Error())
+		return ""
+	}
+	if url != "" {
+		url = strings.Replace(url, "\"", "", -1) // remove double quotes from output
+	}
+	return url
+}
+
+// GetKubeAppServiceEndpoint returns the not versioned endpoint URL of the service's app
+// by using kubectl and returns it as string
+func (ts *TestService) GetKubeAppServiceEndpoint() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	nameNoRevision := strings.Split(ts.Name, ":")[0]
+	serviceName := strings.Split(nameNoRevision, "/")[1]
+	args := []string{"get", "apps",
+		"-n", testingNamespace,
+		fmt.Sprint(serviceName),
+		"-o", `jsonpath="{.status.endpoints[0]}"`}
+	url, err := KubectlCmd(args)
+	if err != nil {
+		ts.T.Fatalf("Failed to get global endpoint url:  %v", err.Error())
+		return ""
+	}
+	if url != "" {
+		url = strings.Replace(url, "\"", "", -1) // remove double quotes from output
+	}
+	return url
+}
+
+// GetAppEndpoint retrieves the not versioned endpoint URL of the service's app
+// by using rio's App object and returns it as string
+func (ts *TestService) GetAppEndpoint() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	return ts.App.Status.Endpoints[0]
```

Comment:
This is good, however it may error out if DNS isn't ready yet. Follow the pattern in `waitForEndpointDNS` and make a `waitForAppEndpointDNS`